### ISSUE TITLE
'BuildDoc' has no such option 'project'

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#831]
I set up **setup_command.BuildDoc** in my **setup.py** as described in its docstring.

```
$ setup.py build_sphinx --help}}}
{{{$ setup.py build_sphinx --help
```

show, among other:

```
...
  --project          The documented project's name
  --version          The short X.Y version
  --release          The full version, including alpha/beta/rc tags
...
```

But when I trying to pass the option **--project=projectname** the script returns an error:

```
running build_sphinx
error: error in command line: command 'BuildDoc' has no such option 'project'
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/831
- Originally reported by: Anonymous
- Originally created at: 2011-11-29T18:22:59.779

**Comment by shimizukawa** [comment]
_From [t_only_ttt](https://bitbucket.org/t_only_ttt) on 2011-12-06 16:03:44+00:00_

I had to do the following to get the project option working

{{{
diff -ur Sphinx-1.1_original/sphinx/setup_command.py Sphinx-1.1/sphinx/setup_command.py
--- Sphinx-1.1_original/sphinx/setup_command.py 2011-11-30 10:07:32.000000000 +0000
+++ Sphinx-1.1/sphinx/setup_command.py  2011-12-06 10:21:53.000000000 +0000
@@ -81,6 +81,7 @@
         self.fresh_env = self.all_files = False
         self.source_dir = self.build_dir = None
         self.builder = 'html'
-        self.project = ''
       self.version = ''
       self.release = ''
       self.today = ''
  @@ -125,6 +126,8 @@
       else:
           status_stream = sys.stdout
       confoverrides = {}
-        if self.project:
-             confoverrides['project'] = self.project  
       if self.version:
            confoverrides['version'] = self.version
       if self.release:
  }}}

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-01-29 11:02:37+00:00_

Closes #831: Provide `--project` flag in setup_command as advertised.

→ <<cset 1bef661b485aed9e3bbb83656c8ac6bc08621ba7>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-10-18 07:32:54+00:00_

Removing version: 1.1.1 (automated comment)
